File: py/dust/test/test-curve.py
from dust.crypto.curve import Keypair, Key
from dust.core.util import encode, decode

keypair=Keypair(Key(decode('48ef7b8c0c6343b332a02862335c11e0cd2b38f0d7d6a3647a7f8f2661d1c946'), False), Key(decode('43aafb64bc96460f3928f6068b2a01aa87bac16da6dc034b4525d1837e9cb85e'), False))
session=keypair.createSessionBytes(decode('77516be967ad1a68632a3c8b08c9e032d23ff5b764a7fe49461b9c12b2da0c32'))
print(encode(session))

File: v1/java/src/dust/test/TestCurve.java
package dust.test;

import java.util.Random;

import dust.crypto.Curve;
import dust.crypto.Keypair;
import dust.crypto.Key;
import dust.core.Util;

public class TestCurve
{
  static public void main(String[] args) throws Exception
  {
    byte[] entropy=new byte[32];
    Random random=new Random();

    random.nextBytes(entropy);
    Keypair pair1=Curve.createKeypair(entropy);
    System.out.println("pair1: "+pair1);

    random.nextBytes(entropy);
    Keypair pair2=Curve.createKeypair(entropy);
    System.out.println("pair2: "+pair2);

    Key session1=pair1.session(pair2.getPublic());
    System.out.println("session1: "+session1);

    Key session2=pair2.session(pair1.getPublic());
    System.out.println("session1: "+session2);
  }
}